saleor/graphql/order/mutations/orders.py

```diff
@@ -560,3 +560,43 @@ def perform_mutation(cls, _root, info, **data):
             cls.get_node_or_error(info, data["id"], only_type=Invoice)
         )
         return super().perform_mutation(_root, info, **data)
+
+
+class UpdateInvoice(ModelMutation):
+    invoice = graphene.Field(Invoice, description="An updated invoice.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of an invoice to update.")
+        number = graphene.String(description="Invoice number")
+        url = graphene.String(description="URL of an invoice to download.")
+
+    class Meta:
+        description = "Updates an invoice."
+        model = models.Invoice
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        if not (instance.number or data.get("number")) or not (
```

Comment:
It is only a visual change, but imo it could increase readbility.
```
number = instance.number or data.get("number")
url = instance.url or data.get("url")
if not number or not url:
```